core-bundle/src/Controller/ContentElement/AbstractContentElementController.php

```diff
@@ -53,5 +59,5 @@ protected function addSharedMaxAgeToResponse(Response $response, ContentModel $m
         $response->setSharedMaxAge(min($min));
     }
 
-    abstract protected function getResponse(Template $template, ContentModel $model, Request $request): Response;
+    abstract protected function getResponse(Template $template, ContentModel $model, Request $request): ?Response;
```

Comment:
Why exactly is this necessary? Because it is a breaking change that will lead to an `Declaration of MyController::getResponse() should be compatible with AbstractContentElementController::getResponse()` exception.

Reply:
It did not for me. Strengthening the requirement (`: Response` instead of `: ?Response`) is always allowed. Same as specifying a return value in your class even though the interface has none.

Reply:
But it weakens the requirement from `Response` to `?Response`  - which does not seem to be necessary anway, since there will be a response now always?

Reply:
No, you don‘t have to return a response anymore. If none is returned, the parent will generate one from the template.

Reply:
> It did not for me.

Then you did it wrong, because it definitely happens: https://3v4l.org/i6uls

Reply:
Hehe, correct. And that's the point. Your controller extends our abstract controller. Your controller is more specific that this one 😉

Reply:
You are absolutely right.